### ISSUE TITLE
SignatureDoesNotMatch if using nginx route based on bucket names

### ISSUE THREAD

**Issue by dingjs** [minio/minio-java#648]
Accoding to the [Setup Nginx proxy with Minio Server](https://docs.minio.io/docs/setup-nginx-proxy-with-minio), we can using nginx route based on bucket names.
If I config nginx like the doc said：
```
location /somebucketName/ {
   proxy_buffering off;
   proxy_set_header Host $http_host;
   proxy_pass http://localhost:9000;
 }
```
The java client would throw SignatureDoesNotMatch exception,but the mc client works fine.
I did some debugging,and find out why.
This is the nginx log
```
"GET /somebucketName?location= HTTP/1.1" 301 185 "
"GET /somebucketName/?location= HTTP/1.1" 403 348 "
```
Because the nginx location is `/somebucketName/`,the `/` suffix did the trick,the java client use `/somebucketName` as the url (no `/` suffix ),but nginx rewrite the url to `/somebucketName/` (with the `/` as suffix),so the client signature does not match the server signature.If I config the location like `/somebucketName` (with no `/` suffix),everything works fine.
This may not be a bug of java client,but is not consistent with mc and the minio docs.
Thanks!

**Comment by vadmeste** [comment]
Thanks for reporting this @dingjs, can you paste the minimal java code here ? I guess it is only initialization and get object, right ?

**Comment by dingjs** [comment]
Yes, I just called minioClient.getObject(bucketName,objectName). We talked about it in slack.

**Comment by balamurugana** [comment]
The fix is changing https://github.com/minio/minio-java/blob/master/api/src/main/java/io/minio/MinioClient.java#L1216 to `HttpResponse response = execute(Method.GET, US_EAST_1, bucketName, "/",` to comply with the spec

**Comment by nitisht** [comment]
@dingjs this was fixed with #650 can you please test with latest `minio-java` master and let us know how it goes

**Comment by dingjs** [comment]
>  this was fixed with #650 can you please test with latest minio-java master and let us know how it goes
@nitisht  I test getObject and putObject using latest minio-java master, it works fine now.
Thanks a lot.
